v2/api/dataprotection/customizations/backup_vaults_backup_instance_extensions.go

```diff
@@ -0,0 +1,158 @@
+
+		// return success for reconcilation if the state is non-retryable
+		if nonRetryableStates.Contains(protectionStatus) {
+			log.V(Debug).Info("Returning PostReconcileCheckResultSuccess")
+			return next(ctx, obj, owner, resolver, armClient, log)
+		}
+
+		// call sync api only when protection status is ProtectionError and error code is usererror
+		if protectionStatus == "ProtectionError" {
+			var protectionStatusErrorCode string
+			protectionStatusErrorCode = strings.ToLower(*backupInstance.Status.Properties.ProtectionStatus.ErrorDetails.Code)
+			log.V(Debug).Info(fmt.Sprintf("Protection Error code is  %q", protectionStatusErrorCode))
+
+			if protectionStatusErrorCode != "" && strings.Contains(protectionStatusErrorCode, "usererror") {
+				id, _ := genruntime.GetAndParseResourceID(backupInstance)
+				subscription := id.SubscriptionID
+				rg := id.ResourceGroupName
+				vaultName := id.Parent.Name
+
+				clientFactory, err := armdataprotection.NewClientFactory(subscription, armClient.Creds(), armClient.ClientOptions())
+				if err != nil {
+					return extensions.PostReconcileCheckResultFailure("failed to create armdataprotection client"), err
+				}
+
+				var parameters armdataprotection.SyncBackupInstanceRequest
+				parameters.SyncType = to.Ptr(armdataprotection.SyncTypeDefault)
+
+				// get the resume token from the resource
+				pollerResumeToken, _ := GetPollerResumeToken(obj, log)
+
+				// BeginSyncBackupInstance is in-progress - poller resume token is available
+
+				log.V(Debug).Info("Starting BeginSyncBackupInstance")
+
+				poller, err := clientFactory.NewBackupInstancesClient().BeginSyncBackupInstance(ctx, rg, vaultName, backupInstance.AzureName(), parameters, &armdataprotection.BackupInstancesClientBeginSyncBackupInstanceOptions{
+					ResumeToken: pollerResumeToken,
+				})
+				if err != nil {
+					return extensions.PostReconcileCheckResultFailure("Failed Polling for BeginSyncBackupInstance to get the result"), err
+				}
+
+				if pollerResumeToken == "" {
+					resumeToken, resumeTokenErr := poller.ResumeToken()
+					if resumeTokenErr != nil {
+						return extensions.PostReconcileCheckResultFailure("couldn't create PUT resume token for resource"), resumeTokenErr
+					} else {
+						SetPollerResumeToken(obj, resumeToken, log)
+					}
+				}
+
+				_, pollErr := poller.Poll(ctx)
+				if pollErr != nil {
+					return extensions.PostReconcileCheckResultFailure("couldn't create PUT resume token for resource"), pollErr
+				}
+
+				if poller.Done() {
+					log.V(Debug).Info("Polling is completed")
+					ClearPollerResumeToken(obj, log)
+					_, err := poller.Result(ctx)
+					if err != nil {
+						return extensions.PostReconcileCheckResultFailure("couldn't create PUT resume token for resource"), err
+					}
+				}
+				log.V(Debug).Info("Polling is in-progress")
+			}
+		}
+	}
+	return extensions.PostReconcileCheckResultFailure("Backup Instance is in non terminal state"), nil
```

Comment:
I can't see anywhere that `PostReconcileCheck` is passing?

Reply:
// return success for reconcilation if the state is non-retryable
		if nonRetryableStates.Contains(protectionStatus) {
			log.V(Debug).Info("Returning PostReconcileCheckResultSuccess")
			return next(ctx, obj, owner, resolver, armClient, log)
		}

Reply:
Yep, I saw that. My concern was mainly, what happens when `protectionStatus` is a success?

Reply:
Reconciler would still do a postReconcileCheck, and be always falling on line 157?